### PR TITLE
Feature/developer to 0.4.0

### PR DESCRIPTION
**Pull request by samleeflang**
Update of developers schema to version openDS 0.4.0

### CODE REVIEW

File: data-model/developer-schema/annotation/0.4.0/schema/annotation-batch-metadata.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  "properties": {
+    "ods:placeInBatch": {
+      "description": "For batching only. Links batch metadata to specific annotations in an event. Value must correspond to an annotation's ods:placeInBatch",
+      "type": "integer"
+    },
+    "searchParams": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
```
Comment (southeo) [inline data-model/developer-schema/annotation/0.4.0/schema/annotation-batch-metadata.json:13]: add `additionalProperties:false` in this declaration

File: data-model/developer-schema/machine-annotation-service/0.4.0/schema/machine-annotation-service-request.json [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        "$ref": "https://schemas.dissco.tech/schemas/fdo-type/shared-model/0.4.0/secret-variable.json"
+      }
+    }
+  },
+  "required": [
+    "schema:name",
+    "ods:containerImage",
+    "ods:containerTag",
+    "ods:batchingPermitted",
+    "ods:TargetDigitalObjectFilter"
```
Comment (southeo) [inline data-model/developer-schema/machine-annotation-service/0.4.0/schema/machine-annotation-service-request.json:177]: `ods:HASTargetDigitalObjectFilter`

File: data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    "ods:hasTombstoneMetadata": {
+      "type": "object",
+      "description": "Object containing the tombstone metadata of the object. Only present when ods:status is ods:Tombstone",
+      "$ref": "https://schemas.dissco.tech/schemas/fdo-type/shared-model/0.4.0/tombstone-metadata.json"
+    }
+  },
+  "required": [
+    "schema:name",
+    "schema:uri",
```
Comment (southeo) [inline data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json:65]: this says `schema:uri`, the above says `schema:url`, which one do we use?

File: data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "ods:translatorType",
+    "ods:mappingID"
```
Comment (southeo) [inline data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json:67]: ods:DATA mappingId

**Review by southeo — COMMENTED** [review]
almost there, just need to make sure the `required` fields are also updated

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -35,24 +35,30 @@ jobs:
           format: 'table'
           ignore-unfixed: true
           vuln-type: 'os,library'
+      - name: Trivy - List all vulnerabilities
```
Comment (southeo) [inline .github/workflows/build.yaml:38]: You have this action twice now, you should remove the other one or the build fails

**Review by southeo — COMMENTED** [review]
one more comment

File: data-model/developer-schema/annotation/0.4.0/schema/annotation-batch-metadata.json (hunk 1 above)
Comment (samleeflang) [reply data-model/developer-schema/annotation/0.4.0/schema/annotation-batch-metadata.json:13]: :+1:

File: data-model/developer-schema/machine-annotation-service/0.4.0/schema/machine-annotation-service-request.json (hunk 2 above)
Comment (samleeflang) [reply data-model/developer-schema/machine-annotation-service/0.4.0/schema/machine-annotation-service-request.json]: :+1:

File: data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json (hunk 4 above)
Comment (samleeflang) [reply data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json]: :+1:

File: data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json (hunk 3 above)
Comment (southeo) [reply data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json]: Still open

File: data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json (hunk 4 above)
Comment (southeo) [reply data-model/developer-schema/source-system/0.4.0/schema/source-system-request.json]: Still open

**Review by southeo — APPROVED** [review]
:kaaba:

**Pull request merged**